modules/kafka/kafka-dest-worker.c

```diff
@@ -61,30 +61,59 @@ _publish_message(KafkaDestWorker *self, LogMessage *msg)
 {
   KafkaDestDriver *owner = (KafkaDestDriver *) self->super.owner;
   int block_flag = _is_poller_thread(self) ? 0 : RD_KAFKA_MSG_F_BLOCK;
-
-  if (rd_kafka_produce(owner->topic,
+  rd_kafka_topic_t *topic;
+  GString *topic_name_new = g_string_sized_new(128);
+  gboolean is_fallback_topic = FALSE;
+  if (owner->topicname_is_a_template)
+    {
+      log_template_format(owner->topic_name, msg, &owner->template_options, LTZ_SEND, self->super.seq_num, NULL,
+                          topic_name_new);
+      topic = _query_topic_from_hash_map(owner, topic_name_new->str);
+      if(!topic)
+        {
+          if(_validate_topic_name(owner, topic_name_new->str)) //Insert into hashtable if topic name is valid
+            {
+              topic = _insert_topic_into_hash_map(owner, topic_name_new->str);
+            }
+        }
+      if(!topic) //if topic name is invalid, then topic is fallback topic which is we know is defined
+        {
+          topic = owner->fallback_topic;
+          is_fallback_topic = TRUE;
+        }
+    }
+  else //Case where topic name isn't a template. Since we have validated the topic name at driver, it is good to eliminate the need for fallback topic
+    {
+      topic = owner->topic;
+      g_string_assign(topic_name_new, owner->topic_name->template);
+    }
+  if (rd_kafka_produce(topic,
                        RD_KAFKA_PARTITION_UA,
                        RD_KAFKA_MSG_F_FREE | block_flag,
                        self->message->str, self->message->len,
                        self->key->len ? self->key->str : NULL, self->key->len,
                        log_msg_ref(msg)) == -1)
     {
       msg_error("kafka: failed to publish message",
-                evt_tag_str("topic", owner->topic_name),
+                owner->topicname_is_a_template ? (!is_fallback_topic ? evt_tag_str("topic", topic_name_new->str) :
+                                                  evt_tag_str("fallback_topic", (owner->fallback_topic_name ? owner->fallback_topic_name : "NULL"))) :
```

Comment:
I would just use the same tag name here with the actual topic we are publishing to. Either the output of the template or the fallback.